### PR TITLE
added structs/methods and some game data impls

### PR DESCRIPTION
**Pull request by triabolicals**
- changed the gamedata get list functions to search for the method in the parent class for SkillData
- changed a field in ProcInst to be mut for a menu to display the title bar correctly when exiting menu
- HubSequence fields appears not be aligned properly (trying to get the scene name directly is invalid)

### CODE REVIEW

File: src/force.rs [hunk 1]
```diff
@@ -17,28 +16,30 @@ pub enum ForceType {
 #[unity::class("App", "Force")]
 pub struct Force {
     pub head: Option<&'static Unit>,
-    pub tail: Option<&'static Unit>,
+	pub tail: Option<&'static Unit>,
```
Comment (Raytwo) [inline src/force.rs:19]: Alignment lol

File: src/gamedata.rs [hunk 2]
```diff
@@ -111,38 +125,103 @@ pub struct WeaponMask {
...
+        let method3 = Self::class()._1.parent.get_methods().iter();
+        for i in method3 {
+            println!("Parent Method: {}", i.get_name().unwrap());
+        }
+        let method = Self::class()._1.parent._1.parent.get_methods().iter();
+        for i in method {
+            println!("Parent Parent Method: {}", i.get_name().unwrap());
+        }
+    }
+}
```
Comment (Raytwo) [inline src/gamedata.rs:227]: Remove this, doesn't belong here

File: src/gamedata/animal.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+impl AnimalData {
+    pub fn can_capture(&self) -> bool { unsafe { is_capture_animal(self, None) }}
+    pub fn increment_amount_capture(&self, amount: i32) { unsafe { inc_animal_capture_num(self, amount, None); }}
+    pub fn increment_capture(&self) { unsafe { inc_animal_capture_num(self, 1, None); }}
+    pub fn get_animal_by_pid(pid: &Il2CppString) -> Option<&AnimalData> { unsafe { get_animal_by_PID(pid, None) } }
+}
+
+#[unity::from_offset("App", "AnimalData", "GetByPID")]
+fn get_animal_by_PID(pid: &Il2CppString, method_info: OptionalMethod) -> Option<&AnimalData>;
```
Comment (Raytwo) [inline src/gamedata/animal.rs:17]: Lowercase for Rust conventions

File: src/gamedata/item.rs [hunk 4]
```diff
@@ -15,6 +17,70 @@ pub struct ItemData {
...
+pub fn UnitItem_IsWeapon(this: &UnitItem, method_info: OptionalMethod) -> bool;
+
+#[unity::from_offset("App", "UnitItem", "get_IsDrop")]
+pub fn UnitItem_get_is_drop(this: &UnitItem, method_info: OptionalMethod) -> bool;
+
+#[unity::from_offset("App", "UnitItem", "GetPower")]
+pub fn UnitItem_GetPower(this: &UnitItem, method_info: OptionalMethod) -> i32;
+
+#[unity::from_offset("App", "UnitItem", "GetEquipSkills")]
+pub fn UnitItem_GetEquipSkills(this: &UnitItem, method_info: OptionalMethod) -> Option<&SkillArray>;
```
Comment (Raytwo) [inline src/gamedata/item.rs:86]: Lowercase for Rust conventions

File: src/gamedata/job.rs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+fn job_get_DiffGrowL(this: &JobData, method_info: OptionalMethod) -> &mut CapabilitySbyte;
+
+#[unity::from_offset("App", "JobData", "get_DiffGrowHard")]
+fn job_get_DiffGrowH(this: &JobData, method_info: OptionalMethod) -> &mut CapabilitySbyte;
+
+#[unity::from_offset("App", "JobData", "get_DiffGrowNormal")]
+fn job_get_DiffGrowN(this: &JobData, method_info: OptionalMethod) -> &mut CapabilitySbyte;
+
+#[unity::from_offset("App", "JobData", "get_DiffGrow")]
+fn job_get_DiffGrow(this: &JobData, method_info: OptionalMethod) -> &mut CapabilitySbyte;
```
Comment (Raytwo) [inline src/gamedata/job.rs:68]: Lowercase

File: src/gamedata/person.rs [hunk 6]
```diff
@@ -59,67 +73,45 @@ impl PersonData {
...
+#[unity::from_offset("App", "PersonData", "set_Fid")]
+fn person_set_fid(this: &PersonData, fid: &Il2CppString, method_info: OptionalMethod);
 
-// PersonData 
 #[unity::from_offset("App", "PersonData", "get_Name")] //#[skyline::from_offset(0x1f25d40)]
 fn person_get_name(this: &PersonData, method_info: OptionalMethod) -> Option<&Il2CppString>;
 
 #[unity::from_offset("App", "PersonData", "get_UnitIconID")] //#[skyline::from_offset(0x1f25d20)]
-fn get_UnitIconID(this: &PersonData, method_info: OptionalMethod) -> &Il2CppString;
+fn get_UnitIconID(this: &PersonData, method_info: OptionalMethod) -> &'static Il2CppString;
```
Comment (Raytwo) [inline src/gamedata/person.rs:114]: Lowercase

File: src/gamedata/unit.rs [hunk 7]
```diff
@@ -45,28 +75,28 @@ pub struct Unit {
...
+    pub m_EquipSkill :&'static SkillArray,
+    pub m_PrivateSkill :&'static SkillArray,
+    pub m_ReceiveSkill :&'static SkillArray,
+    pub m_SupportedSkill :&'static SkillArray,
+    pub m_EquipSkillPool :&'static SkillArray,
+    pub m_LearnedJobSkill :&'static SkillArray,
     m_OriginalAptitude :u64,
-    m_Aptitude :u64,
-    m_WeaponMask :u64,
+    pub m_Aptitude : &'static mut WeaponMask,
```
Comment (Raytwo) [inline src/gamedata/unit.rs:94]: L o w e r c a s e, and no m_

File: src/gamedata/unit.rs [hunk 8]
```diff
@@ -167,3 +226,89 @@ extern "C" fn unit_is_engaging(this: &Unit, method_info: OptionalMethod) -> bool
...
+fn unit_has_private_skill(this: &Unit, sid: &Il2CppString, method_info: OptionalMethod) -> bool;
+
+#[skyline::from_offset(0x01a3a040)]
+fn unit_level_up(this: &Unit, min_stats: i32, method_info: OptionalMethod);
+
+#[unity::from_offset("App","Unit", "SetWeaponMaskFromParson")]
+fn unit_set_weapon_mask_from_person(this: &Unit, method_info: OptionalMethod);
+
+#[unity::from_offset("App", "Unit", "get_Aptitude")]
+fn get_unit_aptitude(this: &Unit, method_info: OptionalMethod) -> &'static mut WeaponMask;
```
Comment (Raytwo) [inline src/gamedata/unit.rs:265]: Same

File: src/gamedata/unit.rs [hunk 9]
```diff
@@ -167,3 +226,89 @@ extern "C" fn unit_is_engaging(this: &Unit, method_info: OptionalMethod) -> bool
 // App.Unit$$IsEngageOwner	7101a197a0	bool App.Unit$$IsEngageOwner(App_Unit_o * __this, MethodInfo * method)	112
 #[skyline::from_offset(0x1a197a0)]
 extern "C" fn unit_is_engage_owner(this: &Unit, method_info: OptionalMethod) -> bool;
+
+//triabolical added
+#[unity::from_offset("App", "Unit", "AddSkillPoint")]
+fn unit_add_SP(this: &Unit, value: i32, method_info: OptionalMethod);
```
Comment (Raytwo) [inline src/gamedata/unit.rs:232]: You know the drill

**Review by Raytwo — CHANGES_REQUESTED** [review]

File: src/gamedata/unit.rs (hunk 7 above)
Comment (triabolicals) [reply src/gamedata/unit.rs:94]: None of the unit field variables are renamed from the original unit.rs. Should probably stay unchanged for capability.

File: src/force.rs (hunk 1 above)
Comment (triabolicals) [reply src/force.rs:19]: That is actually one tab. Not sure why its doubled the size.

**Review by Raytwo — APPROVED** [review]

**Pull request merged**